types/shares.go

```diff
@@ -83,12 +88,10 @@ func splitContiguous(nid namespace.ID, rawDatas [][]byte) []NamespacedShare {
 	innerIndex := 0
 	for outerIndex < len(rawDatas) {
 		var rawData []byte
-		startIndex := 0
-		rawData, outerIndex, innerIndex, startIndex = getNextChunk(rawDatas, outerIndex, innerIndex, TxShareSize)
-		rawShare := append(append(append(
-			make([]byte, 0, len(nid)+1+len(rawData)),
+		rawData, outerIndex, innerIndex, _ = getNextChunk(rawDatas, outerIndex, innerIndex, MsgShareSize)
+		rawShare := append(append(
+			make([]byte, 0, MsgShareSize),
 			nid...),
-			byte(startIndex)),
```

Comment:
Oh, was there a bug in the split logic?

Reply:
Kinda sorta. I still have to iron out the details, but the length delimiter for contiguous Txs was being added twice. Fixing it here caused other problems elsewhere (all the modifications to the tests). We just have to clarify when and where to add the `SHARE_RESERVED_BYTE` and I think were good to go.